### DISCUSSION TITLE
Support GeoIP Module

### DISCUSSION THREAD

**Discussion by nirga** [Kong/kong#7336] [Ideas and feature requests]
### Summary

Understanding the GeoIP data for an incoming request is a common use case for many API GWs. I think Kong should support natively adding the [GeoIP NGINX module](https://docs.nginx.com/nginx/admin-guide/dynamic-modules/geoip/?_ga=2.206236432.2085762362.1612098774-138968329.1612098774). Currently the only way to do that is to create a fork and compile Kong from source.

**Comment by fffonion** [comment]
Hi @nirga, though I agree GeoIP support is a good feature for Kong, do note that the GeoIP NGINX module only works with the "legacy" database format. MaxMind seems to stop the legacy format from publicly available since 2019 and it will be [officially unsupported in around a year](https://blog.maxmind.com/2020/06/01/retirement-of-geoip-legacy-downloadable-databases-in-may-2022/). There're convention tools available but we should probably go with [geoip2 module](https://github.com/leev/ngx_http_geoip2_module) instead, but it's not officially supported by Nginx.

**Comment by Elyahou** [comment]
@fffonion You can find the geoip2 module in [nginx doc](https://www.nginx.com/products/nginx/modules/geoip2/), so it seems to be supported... I created a branch to add support in kong-build-toold: https://github.com/Kong/kong-build-tools/pull/379. What do you think about this ?

**Comment by fffonion** [comment]
@Elyahou Ah `Supported by NGINX for active NGINX Plus subscribers`, interesting, that proves the module works fine in current version of Nginx I suppose 😄 . I don't see a reason we shouldn't do it then, but I have several thoughts:
- Should it be compiled as a dynamic nginx module and not loaded by default, and user can selectively enable it?
- If it's a dynamic module, should we provide it in a seperate binary package, like `kong-geoip2-module` so that `kong` itself doesn't depend on libmaxminddb lib?

**Comment by nirga** [comment]
@fffonion does it make sense to work towards merging [Kong/kong-build-tools#379](https://github.com/Kong/kong-build-tools/pull/379)?

**Comment by codemug** [comment]
Hi @fffonion 
Would be great if this can get some traction, it's a very sought after feature
